tests/neg-custom-args/isInstanceOf/enum-approx2.scala

```diff
@@ -4,5 +4,7 @@ case class Fun[A, B](f: Exp[A => B]) extends Exp[A => B]
 class Test {
   def eval(e: Fun[Int, Int]) = e match {
     case Fun(x: Fun[Int, Double]) => ???          // error
+    case Fun(x: Exp[Int => String]) => ???        // error
+    case _ =>
```

Comment:
Not sure why the 2nd case was removed before, restoring it seems to be more reasonable. Otherwise, it seems not easy to justify why the 1st case receives a warning.

Reply:
The warning on the first one became (and now revert away from) an unchecked warning, on the unchecked type test.